tools/testsys/src/aws_resources.rs

```diff
@@ -342,6 +341,295 @@ impl Migration for AwsK8s {
+        }
+    }
+
+    fn ecs_test_crds(&self, testsys_images: &TestsysImages) -> Result<Vec<Crd>> {
+        let crds = vec![
+            self.ecs_crd(testsys_images)?,
+            self.ec2_crd(testsys_images, None)?,
+            self.ecs_test_crd("-test", None, testsys_images)?,
+        ];
+        Ok(crds)
+    }
+
+    async fn migration_test_crds(&self, testsys_images: &TestsysImages) -> Result<Vec<Crd>> {
+        let ami = self
+            .starting_image_id
+            .as_ref()
+            .unwrap_or(
+                &get_ami_id(
+                    format!(
+                        "bottlerocket-{}-{}-{}-{}",
+                        self.variant, self.arch, self.starting_version.as_ref().context("The starting version must be provided for migration testing")?, self.migrate_starting_commit.as_ref().context("The commit for the starting version must be provided if the starting image id is not")?
+                    ), & self.arch,
+                    self.region.to_string(),
+                )
+                .await?,
+            )
+            .to_string();
+        let ecs = self.ecs_crd(testsys_images)?;
+        let ec2 = self.ec2_crd(testsys_images, Some(ami))?;
+        let mut depends_on = Vec::new();
+        let initial = self.ecs_test_crd("-1-initial", None, testsys_images)?;
+        depends_on.push(initial.name().context("Crd missing name")?);
+        let start_migrate = self.migration_crd(
+            format!("{}-2-migrate", self.cluster_name()),
+            MigrationVersion::Migrated,
+            Some(depends_on.clone()),
+            testsys_images,
+        )?;
+        depends_on.push(start_migrate.name().context("Crd missing name")?);
+        let migrated =
+            self.ecs_test_crd("-3-migrated", Some(depends_on.clone()), testsys_images)?;
+        depends_on.push(migrated.name().context("Crd missing name")?);
+        let end_migrate = self.migration_crd(
+            format!("{}-4-migrate", self.cluster_name()),
+            MigrationVersion::Starting,
+            Some(depends_on.clone()),
+            testsys_images,
+        )?;
+        depends_on.push(end_migrate.name().context("Crd missing name")?);
+        let last = self.ecs_test_crd("-5-final", Some(depends_on.clone()), testsys_images)?;
+        Ok(vec![
+            ecs,
+            ec2,
+            initial,
+            start_migrate,
+            migrated,
+            end_migrate,
+            last,
+        ])
+    }
+
+    /// Labels help filter test results with `testsys status`.
+    fn labels(&self) -> BTreeMap<String, String> {
+        btreemap! {
+            "testsys/arch".to_string() => self.arch.to_string(),
+            "testsys/variant".to_string() => self.variant.to_string(),
+        }
+    }
+
+    fn kube_arch(&self) -> String {
+        self.arch.replace('_', "-")
+    }
+
+    fn kube_variant(&self) -> String {
+        self.variant.replace('.', "")
+    }
+
+    /// Bottlerocket cluster naming convention.
```

Comment:
Nit: Maybe an example of how does that look like? Or is the convention documented somewhere?